sklearn/model_selection/_validation.py

```diff
@@ -1167,7 +1182,13 @@ def learning_curve(estimator, X, y, groups=None,
         train_test_proportions = []
         for train, test in cv_iter:
             for n_train_samples in train_sizes_abs:
-                train_test_proportions.append((train[:n_train_samples], test))
+                if stratify and n_train_samples < n_max_training_samples:
+                    cur_train, useless_test = train_test_split(
+                        train, stratify=y[train], train_size=n_train_samples,
+                        test_size=None, random_state=random_state)
```

Comment:
Ok I'm not sure I understand the connection with shuffle fully. Before this patch, the training data was shuffled once and then growing subsets of the dataset are used. With the patch, they are not growing subsets anymore but can be disjoint, right? Was that intended/am I misreading?
This could be fixed by getting an integer random seed for each cv_iter and then using that for all n_train_samples.

Then this would make sense to require shuffling, though we're shuffling twice then, and could remove the previous shuffle code and just shuffle here, right? Though not sure that actually makes the logic easier.

Reply:
Good questions! Actually, multiple calls to `train_test_split` with the same random seed and growing `train_size`s already produce growing subsets - although this seems to be not mentioned in the contract of this function, so I've added checks for that to `test_learning_curve_with_stratify` - please take a look.
I also removed one excessive shuffle before the loop. In principle, we could try to have just one shuffle for each cv iteration (now we shuflle at each iteration of training set growing inside one cv, i.e. during each call of `train_test_split`), but I don't think that we should do this, because, as you've noted, this would complicate the logic.